test/integration/webhook_token_auth_test.go

```diff
@@ -0,0 +1,232 @@
+	}
+
+	// Get master config
+	masterOptions, err := testserver.DefaultMasterOptions()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer testserver.CleanupMasterEtcd(t, masterOptions)
+
+	// Set up a dummy authenticator server
+	authServer := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.String() {
+		case "/authenticate":
+			if r.Method != "POST" {
+				t.Fatalf("Expected POST to /authenticate, got %s", r.Method)
+			}
+			if err := r.ParseForm(); err != nil {
+				t.Fatalf("Error parsing form POSTed to /token: %v", err)
+			}
+			var tokenPost authentication.TokenReview
+			if err = json.NewDecoder(r.Body).Decode(&tokenPost); err != nil {
+				t.Fatalf("Expected TokenReview structure in POST request: %v", err)
+			}
+			if !reflect.DeepEqual(tokenPost, expectedTokenPost) {
+				t.Fatalf("Expected\n%#v\ngot\n%#v", expectedTokenPost, tokenPost)
+			}
+
+			w.Write([]byte(tokenResponse))
+
+		default:
+			t.Fatalf("Unexpected Token Review request: %v", r.URL.String())
+		}
+	}))
+	cert, err := tls.X509KeyPair(authLocalhostCert, authLocalhostKey)
+	authServer.TLS = &tls.Config{
+		Certificates: []tls.Certificate{cert},
+	}
+	authServer.StartTLS()
+	defer authServer.Close()
+
+	authConfigFile, err := ioutil.TempFile("", "test.cfg")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer os.Remove(authConfigFile.Name())
+	authConfigObj := &clientcmdapiv1.Config{
+		Clusters: []clientcmdapiv1.NamedCluster{
+			{
+				Name: "authService",
+				Cluster: clientcmdapiv1.Cluster{
+					CertificateAuthority: caFile.Name(),
+					Server:               "https://127.0.0.1/authenticate",
+				},
+			},
+		},
+		AuthInfos: []clientcmdapiv1.NamedAuthInfo{
+			{
+				Name: "apiServer",
+				AuthInfo: clientcmdapiv1.AuthInfo{
+					ClientCertificateData: authLocalhostCert,
+					ClientKeyData:         authLocalhostKey,
+				},
+			},
+		},
+		CurrentContext: "webhook",
+		Contexts: []clientcmdapiv1.NamedContext{
+			{
+				Name: "webhook",
+				Context: clientcmdapiv1.Context{
+					Cluster:  "authService",
+					AuthInfo: "apiServer",
+				},
+			},
+		},
+	}
+	authConfigContent, err := configapilatest.WriteYAML(authConfigObj)
+	if err != nil {
+		t.Fatalf("error writing config: %v", err)
+	}
+	if err := ioutil.WriteFile(authConfigFile.Name(), authConfigContent, os.FileMode(0644)); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	masterOptions.AuthConfig.WebhookTokenAuthenticators = []configapi.WebhookTokenAuthenticator{
+		{
+			WebhookTokenAuthnConfigFile: authConfigFile.Name(),
+			WebhookTokenAuthnCacheTTL:   "10s",
+		},
+	}
+
+	// Start server
+	clusterAdminKubeConfig, err := testserver.StartConfiguredMaster(masterOptions)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	clientConfig, err := testutil.GetClusterAdminClientConfig(clusterAdminKubeConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Ex
```

Comment:
External / Example / or .. ?

Reply:
ignore, remnant of my copy/paste from the oidc test